pkg/kubelet/cm/admission/errors.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
```

Comment:
I like the idea of having a dedicated `admission` package, but I think there is a better way to handle this abstraction that doesn't require pulling in `topologymanager`, `cpumanager`, etc. here. It's easier if I just propose a patch on top of this than try and explaining everything in text. I will send you something later today or tomorrow to see what you think.

Reply:
Here it is as a diff to the tip of your PR:
https://github.com/klueska/kubernetes/compare/pr/101432%5E...pr/101432

Much of it is a "revert" of many of the changes you had to make to introduce your original package.

Reply:
It is way nicer indeed. I thought about extending the `Error` interface somehow but I stopped short because I could only think of what would end up to be a poor man alternative of type assertion, and came with the approach I propsed. But your solution avoid that direction completely. I'll switch to your direction.

Reply:
Can you use my `kklues@nvidia.com` email for the attribution please?

Reply:
Sure thing; will push again soon anyway.